### PR TITLE
Fixes grace period in delete

### LINKED ISSUE #15060: grace-period isn't passed through `oc delete --cascade=false`
`oc delete pod --all --grace-period=0 --force --cascade=false` does not set the grace period to zero in its request.   The plumbing is missing here:  https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/delete.go#L244 .

Without this power, when GC is enabled (all the time), you no longer a way to express: "delete it now".

@juanvallejo @fabianofranz

### PR DESCRIPTION
**Pull request by fabianofranz**
**What this PR does / why we need it**:  Fixes `kubectl delete` ignoring `--grace-period`. 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes https://github.com/openshift/origin/issues/15060 found in OpenShift.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/cmd/delete.go [hunk 1]
```diff
@@ -289,7 +289,7 @@ func ReapResult(r *resource.Result, f cmdutil.Factory, out io.Writer, isDefaultD
 	return nil
 }
 
-func DeleteResult(r *resource.Result, out io.Writer, ignoreNotFound bool, shortOutput bool, mapper meta.RESTMapper) error {
+func DeleteResult(r *resource.Result, out io.Writer, ignoreNotFound bool, gracePeriod int, shortOutput bool, mapper meta.RESTMapper) error {
```
Comment (sttts) [inline pkg/kubectl/cmd/delete.go]: the int type is strange. At least document here what it means.

File: pkg/kubectl/cmd/delete.go [hunk 2]
```diff
@@ -302,7 +302,12 @@ func DeleteResult(r *resource.Result, out io.Writer, ignoreNotFound bool, shortO
 
 		// if we're here, it means that cascade=false (not the default), so we should orphan as requested
 		orphan := true
-		return deleteResource(info, out, shortOutput, mapper, &metav1.DeleteOptions{OrphanDependents: &orphan})
+		options := &metav1.DeleteOptions{}
+		if gracePeriod >= 0 {
```
Comment (sttts) [inline pkg/kubectl/cmd/delete.go]: `> 0`  should be enough

**Review by juanvallejo — APPROVED** [review]

**Comment by juanvallejo** [comment]
lgtm, aside from maybe adding a small test

File: pkg/kubectl/cmd/delete.go (hunk 2 above)
Comment (fabianofranz) [inline pkg/kubectl/cmd/delete.go]: In fact 0 is a valid value, only negatives are ignored.

Comment (sttts) [inline pkg/kubectl/cmd/delete.go]: Didn't notice that `DeleteOptions.GracePeriodSeconds` is a pointer. Then it makes a difference indeed.

**Pull request merged**